tools/unlabel

```diff
@@ -0,0 +1,162 @@
+  stdlib/moreLabels.2temp.mli > stdlib/moreLabels.mli
+
+#Fix up with templates in tools/unlabel-patches
+perl -p -e "s/type statistics =/type statistics = Hashtbl\.statistics =/" \
+  stdlib/moreLabels.mli > stdlib/moreLabels.temp.mli
+perl -p -e "s/type \(!'a, !'b\) t/type \(!'a, !'b\) t = \('a, 'b) Hashtbl.t/" \
+  stdlib/moreLabels.temp.mli > stdlib/moreLabels.2temp.mli
+perl -p -e\
+  "s/module Make \(H : HashedType\) : S with type key = H.t\
+/`cat tools/unlabel-patches/1.mli`/" \
+  stdlib/moreLabels.2temp.mli > stdlib/moreLabels.3temp.mli
+perl -p -e\
+  "s/module MakeSeeded \(H : SeededHashedType\) : SeededS with type key = H.t\
+/`cat tools/unlabel-patches/2.mli`/" \
+  stdlib/moreLabels.3temp.mli > stdlib/moreLabels.4temp.mli
+perl -p -e\
+  "s/module Make \(Ord : OrderedType\) : S with type key = Ord.t\
+/`cat tools/unlabel-patches/3.mli`/" \
+  stdlib/moreLabels.4temp.mli > stdlib/moreLabels.5temp.mli
+perl -p -e\
+  "s/module Make \(Ord : OrderedType\) : S with type elt = Ord.t\
+/`cat tools/unlabel-patches/4.mli`/" \
+  stdlib/moreLabels.5temp.mli > stdlib/moreLabels.mli
+
+#Unix
+perl -p -e "$LABREGEX" \
+  otherlibs/unix/unixLabels.mli > otherlibs/unix/unix.temp.mli
+#Tildes
+perl -p -e "$TILDEREGEX" \
+  otherlibs/unix/unix.temp.mli > otherlibs/unix/unix.mli
+
+#Remove type equivalences from unix.mli
+
+#If one name is a prefix of another, must be after it in this list.
+#e.g interval_timer_status and interval_timer
+declare -a arr=("error"
+                "process_status"
+                "wait_flag"
+                "file_descr"
+                "open_flag"
+                "seek_command"
+                "file_kind"
+                "stats"
+                "LargeFile.stats"
+                "access_permission"
+                "dir_handle"
+                "lock_command"
+                "sigprocmask_command"
+                "process_times"
+                "tm"
+                "interval_timer_status"
+                "interval_timer"
+                "passwd_entry"
+                "group_entry"
+                "inet_addr"
+                "socket_domain"
+                "socket_type"
+                "sockaddr"
+                "shutdown_command"
+                "msg_flag"
+                "host_entry"
+                "protocol_entry"
+                "service_entry"
+                "terminal_io"
+                "setattr_when"
+                "flush_queue"
+                "flow_action"
+                "socket_bool_option"
+                "socket_int_option"
+                "socket_optint_option"
+                "socket_float_option"
+                "addr_info"
+                "getaddrinfo_option"
+                "name_info"
+                "getnameinfo_option")
+
+for typ in "${arr[@]}"
+do
+    export typ
+    tmpfile=$(mktemp)
+    perl -p -e 's/ = Unix.$ENV{'typ'}//' otherlibs/unix/unix.mli > ${tmpfile}
+    cat ${tmpfile} > otherlibs/unix/unix.mli
+    rm -f ${tmpfile}
+done
```

Comment:
Can't this be done with `s/type \([a-zA-Z0-9_]\+\) = Unix\.\([A-Z][a-zA-Z0-9_]*\.\)\?\1 =/type \1 =/`?

Reply:
There are 40 entries in that list, and only 40 instances of `= Unix.` in `unixLabels.mli`, so you could do it with an even simpler `s/ = Unix.[^ ]* =/=/`, but I think the stronger matching of the type name with backtracking is safer. If that does work, there's a comment at the top of unxLabels.mli which can be removed.